src/Functions/reinterpretAs.cpp

```diff
@@ -535,29 +535,625 @@ using FunctionReinterpretAsFixedString = FunctionReinterpretAs<DataTypeFixedStri
+    {
+        "Usage example",
+        R"(
+SELECT reinterpretAsUInt32(toFloat32(0.2)) AS x, reinterpretAsFloat32(x)
+        )",
+        R"(
+┌──────────x─┬─reinterpretAsFloat32(x)─┐
+│ 1045220557 │                     0.2 │
+└────────────┴─────────────────────────┘
+        )"
+    }
+    };
+    FunctionDocumentation::IntroducedIn introduced_in_reinterpretAsFloat32 = {1, 1};
+    FunctionDocumentation::Category category_reinterpretAsFloat32 = FunctionDocumentation::Category::TypeConversion;
+    FunctionDocumentation documentation_reinterpretAsFloat32 = {description_reinterpretAsFloat32, syntax_reinterpretAsFloat32, arguments_reinterpretAsFloat32, returned_value_reinterpretAsFloat32, examples_reinterpretAsFloat32, introduced_in_reinterpretAsFloat32, category_reinterpretAsFloat32};
+
+    factory.registerFunction<FunctionReinterpretAsFloat32>(documentation_reinterpretAsFloat32);
+
+    FunctionDocumentation::Description description_reinterpretAsFloat64 = R"(
+Performs byte reinterpretation by treating the input value as a value of type Float64.
+Unlike [`CAST`](#cast), the function does not attempt to preserve the original value - if the target type is not able to represent the input type, the output is meaningless.
+    )";
+    FunctionDocumentation::Syntax syntax_reinterpretAsFloat64 = "reinterpretAsFloat64(x)";
+    FunctionDocumentation::Arguments arguments_reinterpretAsFloat64 = {
+        {"x", "Value to reinterpret as Float64.", {"(U)Int*", "Float*", "Date", "DateTime", "UUID", "String", "FixedString"}}
+    };
+    FunctionDocumentation::ReturnedValue returned_value_reinterpretAsFloat64 = {"Returns the reinterpreted value `x`.", {"Float64"}};
+    FunctionDocumentation::Examples examples_reinterpretAsFloat64 = {
+    {
+        "Usage example",
+        R"(
+SELECT reinterpretAsUInt64(toFloat64(0.2)) AS x, reinterpretAsFloat64(x)
+        )",
+        R"(
+┌───────────────────x─┬─reinterpretAsFloat64(x)─┐
+│ 4596373779694328218 │                     0.2 │
+└─────────────────────┴─────────────────────────┘
+        )"
+    }
+    };
+    FunctionDocumentation::IntroducedIn introduced_in_reinterpretAsFloat64 = {1, 1};
+    FunctionDocumentation::Category category_reinterpretAsFloat64 = FunctionDocumentation::Category::TypeConversion;
+    FunctionDocumentation documentation_reinterpretAsFloat64 = {description_reinterpretAsFloat64, syntax_reinterpretAsFloat64, arguments_reinterpretAsFloat64, returned_value_reinterpretAsFloat64, examples_reinterpretAsFloat64, introduced_in_reinterpretAsFloat64, category_reinterpretAsFloat64};
+
+    factory.registerFunction<FunctionReinterpretAsFloat64>(documentation_reinterpretAsFloat64);
+
+    FunctionDocumentation::Description description_reinterpretAsDate = R"(
+Accepts a string, fixed string or numeric value and interprets the bytes as a number in host order (little endian). It returns a date from the interpreted number as the number of days since the beginning of the Unix Epoch.
```

Comment:
Just: `Reinterprets the input value as a `Date` value (assuming little endian order) which is the number of days since the beginning of the Unix epoch 1970-01-01`.